taichi/program/kernel.cpp

```diff
@@ -319,7 +324,12 @@ float64 Kernel::get_ret_float(int i) {
 }
 
 int64 Kernel::get_ret_int(int i) {
-  auto dt = rets[i].dt->get_compute_type();
+  Type *dt;
+  if (auto tensor_type = rets[0].dt->cast<TensorType>()) {
```

Comment:
I think it's problematic to change `i` to 0 here. What about only abstracting the if-else for fetching results out?

Reply:
fix